code/lib/core-common/src/presets.ts

```diff
@@ -106,9 +91,9 @@ export const resolveAddonName = (
   // We don't want to resolve an import path (e.g. '@addons/foo/preview') to the file on disk,
   // because you are not allowed to import arbitrary files in packages in Vite.
   // Instead we check if the export exists and "absolutize" it.
-  const managerFile = absolutizeExport(`/manager`);
-  const registerFile = absolutizeExport(`/register`) || absolutizeExport(`/register-panel`);
-  const previewFile = absolutizeExport(`/preview`);
```

Comment:
We no longer need to turn these into absolute paths in the server, since we are no longer introducing addons that aren't in the user's package.json.  Additionally, this leaves it up to the consumer how to import/require the files, which means they'll get the right file (esm or cjs).

Reply:
Is this not a breaking change for any addons that have not made the changes we've made here in essentials?

> Additionally, this leaves it up to the consumer how to import/require the files, which means they'll get the right file (esm or cjs).

Can we keep this benefit without losing the absolute path (and keep stripping the absolute path in the vite builder?)

Reply:
For anyone else watching, Tom and I chatted about this, and came up with a few ideas I'll try out of how we can do this in a non-breaking way, though other existing "super-addons" would still be broken in the vite builder, most likely, and we may need to log a warning in those cases, if possible.

Reply:
OK, I've turned the `previewAnnotations` of sub-addons into a tuple, containing both the bare import name (for vite) and the absolute path (for webpack and back-compat).  I also added a warning when we can detect that sub-addons are not being re-exported from the top-level addon.  It doesn't have any instructions for how to address it, but my hope is that if it comes up, either users or authors will ask in GitHub/Discord about it.  And once we decide on a final path forward for these kinds of addons, we can add a note in the migration docs, and point to that.